### PR TITLE
Use ES6 instead of CoffeeScript

### PR DESCRIPTION
**Pull request by joker-777**
* converts all files to es6
* adds eslint for linting
* adds babel processing of ES6 modules to building process
* also adds comonjs build which is used by node
* improves the jest tests based on the eslint rules
  * expects `it` instead of `test`
  * expects `expect.assertions` for each test
  * expect a `describe` as the main wrapper

### CODE REVIEW

**Comment by joker-777** [comment]
@gingerlime a small Christmas present ;)

File: src/storage.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// a thin wrapper around localStorage checking if localStorage is available
+class Storage {
+  constructor(namespace = "alephbet") {
+    this.namespace = namespace
+    try {
+      const check = "localstorage_check"
+      localStorage.setItem(check, check)
+      localStorage.removeItem(check)
+      this.storage = JSON.parse(localStorage[this.namespace] || "{}")
```
Comment (gingerlime) [inline src/storage.js:9]: are you going to apply the change from #39 here?

**Review by gingerlime — APPROVED** [review]
obviously impossible to compare all the code, but LGTM 👍

I wonder if we should also update some of the README / Wiki code to ES6 ?

**Comment by gingerlime** [comment]
@joker-777 you can now release new versions to npm by yourself if you want. You just need to update the CHANGELOG and version in package.json and then create a [new release / tag](https://github.com/Alephbet/alephbet/releases) and there's a github action that should release it to npm automatically.

**Comment by joker-777** [comment]
@gingerlime Just want to add that it has a breaking change. This doesn't work anymore

```js
import Alephbet from "alephbet"

new Alephbet.Experiment()
```

but instead you have to writ

```js
import {Experiment} from "alephbet"

new Experiment()
```

The following should still work though

```js
const Alephbet = require("alephbet")

new Alephbet.Experiment()
```
and
```html
<script src="alephbet.min.js"></script>
<script>
new Alephbet.Experiment()
</script>

**Comment by gingerlime** [comment]
seems fine to me, but perhaps it's worth adding a note to the README ?

**Comment by joker-777** [comment]
Yes, I was thinking the same. I will create another PR with some changes for it.

**Pull request merged**